### PR TITLE
Fix setsquare activation and deactivation, fix toolRange

### PR DESCRIPTION
**Pull request by rolandlo**
Preparation for #4443 
The rendering issues when activating or deactivating the setsquare fixed in this PR are fully visible when combined with #4502

### CODE REVIEW

**Comment by rolandlo** [comment]
Merging in 24 hours if no objections are raised.

File: src/core/view/SetsquareView.cpp [hunk 1]
```diff
@@ -45,6 +46,9 @@ SetsquareView::~SetsquareView() noexcept { this->unregisterFromPool(); };
 void SetsquareView::on(FlagDirtyRegionRequest, const Range& rg) { this->parent->flagDirtyRegion(rg); }
 
 void SetsquareView::on(UpdateValuesRequest, double h, double rot, cairo_matrix_t m) {
+    if (h <= 0.) {
+        g_warning("Non-positive setsquare height: %f", h);
+    }
```
Comment (bhennion) [inline src/core/view/SetsquareView.cpp:51]: Does this actually happen? Should this be asserted?

File: src/core/model/Setsquare.cpp [hunk 2]
```diff
@@ -27,7 +30,7 @@ auto Setsquare::getToolRange(bool transformed) const -> Range {
         rg.addPoint(-h, 0);
         rg.addPoint(0, h);
     }
-    rg.addPadding(.5 * xoj::view::SetsquareView::LINE_WIDTH);  // account for line width
+    rg.addPadding(.5 * xoj::view::SetsquareView::LINE_WIDTH * CM);  // account for line width
```
Comment (bhennion) [inline src/core/model/Setsquare.cpp:33]: Could you make the unit of SetsquareView::LINE_WIDTH explicit? e.g. add a comment at the declaration point, and maybe rename it to LINE_WIDTH_IN_CM

**Review by bhennion — APPROVED** [review]
LGTM

File: src/core/model/Setsquare.cpp (hunk 2 above)
Comment (rolandlo) [reply src/core/model/Setsquare.cpp]: I renamed it. A comment doesn't look necessary with the renamed constant.

File: src/core/view/SetsquareView.cpp (hunk 1 above)
Comment (rolandlo) [reply src/core/view/SetsquareView.cpp]: No, I don't think it ever happens. I changed it to an assertion.

**Pull request merged**